testing/binaries/tests/tckconvert

```diff
@@ -1,8 +1,15 @@
-tckconvert tracks.tck tmp.vtk -force && awk '/LINES/,0' tmp.vtk > tmplines1.txt && awk '/LINES/,0' tckconvert/out0.vtk > tmplines2.txt && diff tmplines1.txt tmplines2.txt
-tckconvert tracks.tck -scanner2voxel dwi.mif tmp.vtk -force && awk '/POINTS/{s=1;next}/LINES/{s=0}s' tmp.vtk > tmppoints1.txt && awk '/POINTS/{s=1;next}/LINES/{s=0}s' tckconvert/out1.vtk > tmppoints2.txt && testing_diff_matrix tmppoints1.txt tmppoints2.txt -abs 1e-4
-tckconvert tracks.tck -scanner2voxel dwi.mif tmp.vtk -force && awk '/LINES/,0' tmp.vtk > tmplines1.txt && awk '/LINES/,0' tckconvert/out1.vtk > tmplines2.txt && diff tmplines1.txt tmplines2.txt
+tckconvert -ascii tracks.tck tmp.vtk -force && awk '/POINTS/{s=1;next}/LINES/{s=0}s' tmp.vtk > tmppoints1.txt && awk '/POINTS/{s=1;next}/LINES/{s=0}s' tckconvert/out0.vtk > tmppoints2.txt && testing_diff_matrix tmppoints1.txt tmppoints2.txt -abs 1e-4
+tckconvert tracks.tck tmp.vtk -force
+tckconvert -ascii tracks.tck tmp.vtk -force && awk '/LINES/,0' tmp.vtk > tmplines1.txt && awk '/LINES/,0' tckconvert/out0.vtk > tmplines2.txt && diff tmplines1.txt tmplines2.txt
+tckconvert -ascii tracks.tck -scanner2voxel dwi.mif tmp.vtk -force && awk '/POINTS/{s=1;next}/LINES/{s=0}s' tmp.vtk > tmppoints1.txt && awk '/POINTS/{s=1;next}/LINES/{s=0}s' tckconvert/out1.vtk > tmppoints2.txt && testing_diff_matrix tmppoints1.txt tmppoints2.txt -abs 1e-4
+tckconvert -ascii tracks.tck -scanner2voxel dwi.mif tmp.vtk -force && awk '/LINES/,0' tmp.vtk > tmplines1.txt && awk '/LINES/,0' tckconvert/out1.vtk > tmplines2.txt && diff tmplines1.txt tmplines2.txt
 tckedit tracks.tck -number 10 tmp.tck -nthread 0 && tckconvert tmp.tck tmp-[].txt && cat tmp-*.txt > tmp-all.txt && testing_diff_matrix tmp-all.txt tckconvert/out2-all.txt -abs 1e-4
 tckconvert tckconvert/out2-[2:9].txt tmp.tck -force && testing_diff_tck tmp.tck tckconvert/out3.tck
 echo 1 2 3 > tmp.txt && tckconvert -force -quiet tmp.txt tmp.tck && tckconvert -quiet -force tmp.tck tmp.rib && [ $(wc -l < tmp.rib ) == 4 ]
 tckconvert -force -quiet tckconvert/empty.vtk tmp.tck
+
+# convert tck -> vtk -> txt and compare to tck -> txt
+tckconvert tracks.tck tmp.vtk -binary -force -quiet && tckconvert tmp.vtk tmp-vtk-streamlines-[].txt -force && cat tmp-vtk-streamlines-*.txt > tmp-vtk-all.txt && tckconvert tracks.tck tmp-mrtrix-streamlines-[].txt -force -quiet && cat tmp-mrtrix-streamlines-*.txt > tmp-mrtrix-all.txt && testing_diff_matrix tmp-vtk-all.txt tmp-mrtrix-all.txt
+
+# convert tck -> vtk -> txt and compare to tck -> vtk -> tck -> txt
+tckconvert tracks.tck tmp.vtk -binary -force -quiet && tckconvert tmp.vtk tmp-roundtrip.tck -force && tckconvert tmp-roundtrip.tck tmp-roundtrip-streamlines-[].txt -force && cat tmp-roundtrip-streamlines-*.txt > tmp-roundtrip-all.txt && tckconvert tracks.tck tmp-mrtrix-streamlines-[].txt -force -quiet && cat tmp-mrtrix-streamlines-*.txt > tmp-mrtrix-all.txt && testing_diff_matrix tmp-roundtrip-all.txt tmp-mrtrix-all.txt
```

Comment:
One lesson I've learned the hard way elsewhere is that if your own software is non-compliant, but the reader and writer functionalities have matching non-compliance, then round-trip tests do you no good.

If we could add an exemplar binary file into the test_data repository that demonstrably works in some other software, then we could ensure that that at least gets read correctly. Writing back to binary and then doing a binary diff *might* work for testing the write.

I don't think it's super critical here given that we presume you've implemented the writer for a purpose and it's being successfully read by something else, and also that the code is relatively independent of everything else. But as with another comment, for #411 we'd need to be more exhaustive.

Reply:
Indeed. I'll add the outputs produced by the conversion to the test data repo and test against that. It'll catch regressions, and if @blezek can double-check that these work correctly in other packages, we should be good.

Reply:
I don't have anything with which to visualise polylines.